rmgpy/molecule/fragment.py

```diff
@@ -123,6 +123,799 @@ def is_non_hydrogen(self):
+                        initial_map[atom] = atom_label_map[0]
+                    else:
+                        keys.append(atom)
+                        atms.append(atom_label_map)
+            if atms:
+                for atmlist in itertools.product(*atms):
+                    # skip entries that map multiple graph atoms to the same subgraph atom
+                    if len(set(atmlist)) != len(atmlist):
+                        continue
+                    for i,key in enumerate(keys):
+                        initial_map[key] = atmlist[i]
+                    if self.is_mapping_valid(other, initial_map, equivalent=False) and \
+                            Graph.is_subgraph_isomorphic(self, other, initial_map, save_order=save_order):
+                        return True
+                else:
+                    return False
+            else:
+                if not self.is_mapping_valid(other, initial_map, equivalent=False):
+                    return False
+
+        # Do the isomorphism comparison
+        new_initial_map = None
+        if initial_map:
+            new_initial_map = {}
+            for fragment_vertex in initial_map:
+                repr_mol_vertex = mapping[fragment_vertex]
+                new_initial_map[repr_mol_vertex] = initial_map[fragment_vertex]
+
+        result = Graph.is_subgraph_isomorphic(self.mol_repr, other, new_initial_map)
+        return result
+
+    def get_molecular_weight(self):
+        """
+        Return the fragmental weight of the fragment in kg/mol.
+        """
+        mass = 0
+        for vertex in self.vertices:
+            if isinstance(vertex, Atom):
+                mass += vertex.element.mass
+        return mass
+
+    def calculate_cp0(self):
+        """
+        Return the value of the heat capacity at zero temperature in J/mol*K.
+        """
+        self.assign_representative_molecule()
+        # currently linear fragment will be treated as non-linear molecule
+        return self.mol_repr.calculate_cp0()
+
+    def calculate_cpinf(self):
+        """
+        Return the value of the heat capacity at infinite temperature in J/mol*K.
+        """
+        self.assign_representative_molecule()
+        # currently linear fragment will be treated as non-linear molecule
+        return self.mol_repr.calculate_cpinf()
+
+    def is_radical(self):
+        """
+        Return ``True`` if the fragment contains at least one radical electron,
+        or ``False`` otherwise.
+        """
+        for vertex in self.vertices:
+            if isinstance(vertex, Atom) and vertex.radical_electrons > 0:
+                return True
+        return False
+
+    def update(self, sort_atoms=True):
+        # currently sort_atoms does not work for fragments
+        for v in self.vertices:
+            if isinstance(v, Atom):
+                v.update_charge()
+
+        self.update_atomtypes()
+        self.update_multiplicity()
+        self.sort_vertices()
```

Comment:
The molecule code makes a call to `self.identify_ring_membership()`. Is this needed here?